sebs/aws/aws.py

```diff
@@ -157,7 +166,145 @@ def package_code(
+        auth_data = auth['authorizationData'][0]
+        token = base64.b64decode(auth_data['authorizationToken']).decode('utf-8')
+        username, password = token.split(':')
+        registry_url = auth_data['proxyEndpoint']
+
+        self.docker_client.login(username=username, password=password, registry=registry_url)
+        ret = self.docker_client.images.push(
+            repository=repository_uri, tag=image_tag, stream=True, decode=True
+        )
+        for val in ret:
+            if "error" in val:
+                self.logging.error(f"Failed to push the image to registry {repository_uri}")
+                raise RuntimeError(val)
+
+    def repository_exists(self, repository_client, repository_name):
+        try:
+            repository_client.describe_repositories(repositoryNames=[repository_name])
+            return True
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'RepositoryNotFoundException':
+                return False
+            else:
+                self.logging.error(f"Error checking repository: {e}")
+                raise e 
+
+    def create_repository(self, repository_client, repository_name):
+        if not self.repository_exists(repository_client, repository_name):
+            try:
+                repository_client.create_repository(repositoryName=repository_name)
+                self.logging.info(f"Created ECR repository: {repository_name}")
+            except ClientError as e:
+                if e.response['Error']['Code'] != 'RepositoryAlreadyExistsException':
+                    self.logging.error(f"Failed to create ECR repository: {e}")
+                    raise e
+
+
+    def build_base_image(
+        self,
+        directory: str,
+        language_name: str,
+        language_version: str,
+        benchmark: str,
+        is_cached: bool,
+    ) -> bool:
+        """
+        When building function for the first time (according to SeBS cache),
+        check if Docker image is available in the registry.
+        If yes, then skip building.
+        If no, then continue building.
+
+        For every subsequent build, we rebuild image and push it to the
+        registry. These are triggered by users modifying code and enforcing
+        a build.
+        """
+
+        account_id = self.config.credentials.account_id
+        region = self.config.region
+        registry_name = f"{account_id}.dkr.ecr.{region}.amazonaws.com"
+        # PK: To Do: Get repository name from the config.
+        repository_name = "test_repo"
+        image_tag = self.system_config.benchmark_image_tag(
+            self.name(), benchmark, language_name, language_version
+        )
+        repository_uri = f"{registry_name}/{repository_name}:{image_tag}"
+
+        ecr_client = boto3.client('ecr', region_name=region)
+
+        # Create repository if it does not exist
+        self.create_repository(ecr_client, repository_name)
```

Comment:
We could put all creation of ECR resources inside `config.py` - `AWSResources` class.

Reply:
Yes, the resource creation part is moved to AWSResources class